File: JeongEon8/01/[PGS - Lv2] 점프와 순간 이동.cpp
#include <iostream>

using namespace std;

int solution(int n) {
    int ans = 0;
    
    do {
        if(n%2 != 0) {
            ans++;
            n -= 1;
        }
        else {
            n /= 2;
        }
    } while(n > 0);
    
    return ans;
}

File: 2026 08월/8월 2주차/breadbirds/02/[PGS - Lv2] 점프와 순간 이동.java
public class Solution {
    public int solution(int n) {
        int ans = 0;

        while (n > 0) {
            // 짝수면 순간이동을 거꾸로 한 것
            if (n % 2 == 0) {
                n /= 2;
            }
            // 홀수면 점프를 거꾸로 한 것
            else {
                n -= 1;
                ans++;
            }
        }

        return ans;
    }
}
